hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java

```diff
@@ -1371,6 +1395,13 @@ private boolean getChildrenKeys(String keyPrefix, String startKey,
 
         keysResultList.add(ozoneKey);
 
+        // If shallow is true, all immediate children will be added to
+        // keysResultList. Because only immediate children of keyPrefix
+        // are needed, the stack is always empty.
+        if (isShallow()) {
```

Comment:
If it's shallow list, do we need to execute the logic started from Line 1415?

Reply:
> If it's shallow list, do we need to execute the logic started from Line 1415?

@ChenSammi Thanks for review. No need to execute `reachedLimitCacheSize` logic. Two cases:
1.  **Not reach `reachedLimitCacheSize`**: The result just need execute one time `getNextListOfKeys` -> `getChildrenKeys`. On the second execution, because the last key is used as `startKey`, only its own is returned and will be removed. So this time it's empty result and will exit `getNextListOfKeys`.
2. **Reach `reachedLimitCacheSize`**: Each round of execution uses lastKey from the previous round as the `startKey`. Exit only when  the same as case1.

The logic of reachedLimitCacheSize in the original code is for `file case` only, because need to record startKey. (The startKey of the `folder case` is always "".) This only take effect when `while (!stack.isEmpty())` case.